macOS/debugger.cpp

```diff
@@ -481,7 +481,11 @@ void *Debugger::RemoteAllocateNear(uint64_t region_min,
   //try after first
   min_address = region_max;
   max_address = (UINT64_MAX - region_min < 0x80000000) ? UINT64_MAX : region_min + 0x80000000;
+#ifdef ARM64
+  void *ret_address = RemoteAllocate(size, protection);
```

Comment:
If RemoteAllocate fails and ret_address is NULL, the code will proceed to RemoteAllocateBefore/RemoteAllocateAfter which doesn't make sense on ARM. I think this code would be cleaner if there was a completely separate branch for ARM.

Reply:
technically, if RemoteAllocate fails, it ends in FATAL but you are right, having a separate function is cleaner.